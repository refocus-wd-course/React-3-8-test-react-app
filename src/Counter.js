import React, { useState } from 'react';


function Counter() {
    const [counter, setCounter] = useState(0);

    function handleIncrement() {
        setCounter(counter + 1);
    }

    function handleDecrement() {
        setCounter(counter - 1);
    }

    return (
        <div>
            <label>Counter:</label>
            <p data-testid="counter">{counter}</p>
            <div>
                <button onClick={handleIncrement}>+</button>
                <button onClick={handleDecrement}>-</button>
            </div>
        </div>
    );
}

export default Counter;