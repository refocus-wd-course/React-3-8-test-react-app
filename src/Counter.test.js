import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

test('renders', () => {
    render(<Counter />);
});

test('snapshot test', () => {
    const counter = renderer.create(<Counter />).toJSON();
    expect(counter).toMatchSnapshot();
});

test('increments', () => {
    render(<Counter />);

    const counterValue = screen.getByTestId('counter');
    expect(counterValue).toBeDefined();
    let counter = parseInt(counterValue.innerHTML);

    // console.log('counter before clicking:', counter);
    
    const button = screen.getByText('+');
    expect(button).toBeDefined();

    fireEvent.click(button);

    counter = parseInt(counterValue.innerHTML);

    // console.log('counter after clicking:', counter);

    expect(counter).toBeGreaterThan(0);
});

test('decrements', () => {
    render(<Counter />);

    const counterValue = screen.getByTestId('counter');
    expect(counterValue).toBeDefined();
    let counter = parseInt(counterValue.innerHTML);

    // console.log('counter before clicking:', counter);
    
    const button = screen.getByText('-');
    expect(button).toBeDefined();

    fireEvent.click(button);

    counter = parseInt(counterValue.innerHTML);

    // console.log('counter after clicking:', counter);

    expect(counter).toBeLessThan(0);
});