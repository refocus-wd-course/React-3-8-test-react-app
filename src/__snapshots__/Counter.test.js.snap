// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <label>
    Counter:
  </label>
  <p
    data-testid="counter"
  >
    0
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
